### PR TITLE
Bug 1810743 - Expose detect-only setting in AC

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1810743

### CODE REVIEW

File: android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt [hunk 1]
```diff
@@ -666,6 +666,16 @@ class GeckoEngine(
                 field = value
             }
 
+        override var cookieBannerHandlingDetectOnlyMode: Boolean = false
```
Comment (zmckenney) [inline android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt:669]: There is a `CookieBannerHandlingMode.DETECT_ONLY` so why is this a boolean with a runtime setting for detect only mode? It seems like it should be set the same as the other modes.

**Comment by Amejia481** [comment]
🔴Requires https://bugzilla.mozilla.org/show_bug.cgi?id=1810742 🔴

**Comment by Amejia481** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1810742 🛩️   landed, this is ready to go!

File: docs/changelog.md [hunk 2]
```diff
@@ -9,6 +9,10 @@ permalink: /changelog/
 * [Gecko](https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/plugins/dependencies/src/main/java/Gecko.kt)
 * [Configuration](https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/.config.yml)
 
+* **concept-engine**
+  * 🆕 Added `Settings.cookieBannerHandlingDetectOnlyMode` helps to dected cookie banner events without handle the banners + indicating the mode of the events, see [bug #1810743](https://bugzilla.mozilla.org/show_bug.cgi?id=1810743)
```
Comment (gabrielluong) [inline docs/changelog.md:13]:
```suggestion
  * 🆕 Added `Settings.cookieBannerHandlingDetectOnlyMode` which helps to detect cookie banner events without handle the banners + indicating the mode of the events, see [bug 1810743](https://bugzilla.mozilla.org/show_bug.cgi?id=1810743)
```

**Review by gabrielluong — APPROVED** [review]

**Pull request merged**